Lib/xmlrpc/server.py

```diff
@@ -772,23 +773,13 @@ def docroutine(self, object, name, mod=None,
             self.escape(anchor), self.escape(name))
 
         if inspect.ismethod(object):
-            args = inspect.getfullargspec(object)
+            sig = signature(object)
             # exclude the argument bound to the instance, it will be
             # confusing to the non-Python user
-            argspec = inspect.formatargspec (
-                    args.args[1:],
-                    args.varargs,
-                    args.varkw,
-                    args.defaults,
-                    annotations=args.annotations,
-                    formatvalue=self.formatvalue
-                )
+            sig.replace(parameters=tuple(sig.parameters.values())[1:])
```

Comment:
`sig.replace()` returns a new `Signature` object, so you should have written `sig = sig.replace(...)`.  But `inspect.signature` should take care of the bound argument automatically, so you don't need this dance. `if` and `elif` clauses should be replaced with

```python
if callable(object):
    argspec = str(signature(object))
else:
    argspec = '(...)'
```

Reply:
How common a failure of `signature()`? Shouldn't we catch some common exceptions?

Reply:
`inspect.getfullargspec` uses `inspect.signature` behind the scenes.  So if you want to add some error handling, it should be in a separate PR.